src/pml_vqvae/vqvae/vqvae.py

```diff
@@ -222,5 +226,24 @@ def visualize_output(batch, output, target, prefix: str = "", base_dir: str = ".
             outfile=os.path.join(base_dir, f"{prefix}_reconstruction.png"),
         )
 
+    def set_prior(self, prior: PML_model):
+        self.prior = prior
+
+    def sample(self, class_idx: torch.Tensor):
+        """
+        Sample from the model using the prior
+        :param class_idx: the class index
+        :return: the sample
+        """
+
+        if self.prior is None:
+            raise ValueError("No prior set. Use set_prior() to set a prior model")
+
+        latents = self.prior.sample(class_idx)
+
+        output = self.decoder_stack(latents)
+
+        return output
+
```

Comment:
ich würd das nicht in das model integrieren, benutzen ja das gleiche VQVAE für alle klassen. Glaube ne funktion, oder ne klasse, die nen pixelcnn und n vqvae als input nimmt, und dann daraus bilder samplet sollte genügen. Wir benutzen die funktion ja nicht n der training pipeline

Reply:
Geschmacks Sache. Beides möglich